lib/winapi/ws2_32/winsock.c

```diff
@@ -0,0 +1,232 @@
+#define DHCP
+
+
+#include "lwip/sockets.h"
+#include "lwip/sys.h"
+#include "lwip/netdb.h"
+#include <lwip/debug.h>
+#ifdef DHCP
+#include <lwip/dhcp.h>
+#else
+#include <lwip/autoip.h>
+#endif
+#include <lwip/init.h>
+#include <lwip/netif.h>
+#include <lwip/sys.h>
+#include <lwip/tcpip.h>
+#include <lwip/timeouts.h>
+#include <netif/etharp.h>
+#include <pktdrv.h>
+#include <xboxkrnl/xboxkrnl.h>
+
+#include <hal/debug.h>
+
+#include "winsock2.h"
+
+
+#define PKT_TMR_INTERVAL 5 /* ms */
+#define DEBUGGING        0
+
+struct netif nforce_netif, *g_pnetif;
+
+err_t nforceif_init(struct netif *netif);
+static void packet_timer(void *arg);
+
+static void tcpip_init_done(void *arg)
+{
+    sys_sem_t *init_complete = arg;
+    sys_sem_signal(init_complete);
+}
+
+static void packet_timer(void *arg)
+{
+    LWIP_UNUSED_ARG(arg);
+    Pktdrv_ReceivePackets();
+    sys_timeout(PKT_TMR_INTERVAL, packet_timer, NULL);
+}
+
+#include <windows.h>
+#include <assert.h>
+
+static int map_error_to_winsock(int error) {
+  switch(error) {
+#define MAP_ERROR(error_name) case error_name: return WSA ## error_name;
+#undef MAP_ERROR
+  default:
+    debugPrint("unhandled error: %d\n", error);
+    assert(0);
+    break;
+  }
+  return -1;
+}
+
+
+int WSAStartup(
+  WORD      wVersionRequired,
+  LPWSADATA lpWSAData
+) {
+  assert(wVersionRequired == MAKEDWORD(1,1));
+  lpWSAData->wVersion = wVersionRequired;
+  lpWSAData->wHighVersion = MAKEDWORD(1,1);
```

Comment:
```suggestion
  assert(wVersionRequired == MAKEDWORD(2,0));
  lpWSAData->wVersion = wVersionRequired;
  lpWSAData->wHighVersion = MAKEDWORD(2,0);
```

2.0 is what is used my mingw winsock2.h.